INSTALL.md

```diff
@@ -71,3 +71,22 @@ uvicorn optimade.server.main_index:app --reload --port 5001
 ```
 
 will run the index meta-database server at <http://localhost:5001/v1>.
+
+## Testing specific backends
+
+In order to run the test suite for a specific backend, the `OPTIMADE_DATABASE_BACKEND` [environment variable (or config option)](https://www.optimade.org/optimade-python-tools/configuration/) can be set to one of `'mongodb'`, `'mongomock'` or `'elastic'` (see [`optimade.server.config.ServerConfig.database_backend`](https://www.optimade.org/optimade-python-tools/api_reference/server/config/#optimade.server.config.ServerConfig.database_backend).
+Tests for the two "real" database backends, MongoDB and Elasticsearch, require a writable, temporary database to be accessible.
+The easiest way to deploy these databases and run the tests is with Docker, as shown below:
+
+```shell
+# Run tests on containerized Elasticsearch v6.8 instance
+docker container stop elasticsearch_test; docker container rm elasticsearch_test; \
+    docker run -d --name elasticsearch_test -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" elasticsearch:6.8.13 \
+    && sleep 10
+    && OPTIMADE_DATABASE_BACKEND="elastic" py.test
+
+# Run tests against a real containerized MongoDB
+docker container stop mongo_test; docker container rm mongo_test; \
```

Comment:
1 Perhaps you could clarify a bit what these steps do?
It seems a bit strange that you try to stop the container before you start it.

2 I had to run the commands with sudo privileges.

3 For me the complete test hangs on "test_structures.py". If I do the complete test outside docker the same error occurs, so I do not think it depends on your instructions. Strangely enough, running test_structures.py individually does pass all the tests. I was thus not able to complete the docker instructions.

Reply:
> 1 Perhaps you could clarify a bit what these steps do?
> It seems a bit strange that you try to stop the container before you start it.

This is just to provide a single repeatable command, the docker image needs to be destroyed either before/after testing as otherwise it will contain the test data. Perhaps it makes more sense to have it at the end.

> 2 I had to run the commands with sudo privileges.

This is quite system-dependent and I would prefer to not ask people to use sudo by default (though we can link to the docker documentation); on Arch, installing docker automatically adds the user to the `docker` group so you can run without sudo. 

> 3 For me the complete test hangs on "test_structures.py". If I do the complete test outside docker the same error occurs, so I do not think it depends on your instructions. Strangely enough, running test_structures.py individually does pass all the tests. I was thus not able to complete the docker instructions.

Strange... could you post the full output (maybe with py.test -vvv too?)

> With the elastic search the test did complete normally although there seemed to be fewer tests than for Mongodb.

There is a set of tests that are run for both backends, and some that can only be run on one or the other, so this is fine.

Reply:
>     1. When there are line breaks, the terminal thinks you have pressed enter and executes the statement.
>        https://askubuntu.com/questions/377259/stop-terminal-auto-executing-when-pasting-a-command
>        This is usually not what you want. As you still want to be able to modify the statement.

The continuation characters (`/`) should prevent this (or at least do for me under sh, bash, zsh and fsh...). I did spot a missing `/` in the elastic instructions.

> 2 You are right It is better not to use sudo by default.
> 
> 3 It seems to hang in models/test_structures.py::test_bad_structures it also hangs when I run only test_structures.py (There are two files named test_structures hence the confusion.) I was still running it using my code branch. So perhaps my changes in mongo.py broke something. I'll look into this tomorrow.

These lines basically follow what we do in the CI, so they should work everywhere on the latest commit.

I'll expand a bit on the comments to make clear what we are doing, and perhaps will reorganize the calls so that the images are deleted after the tests, not before the next tests.

Reply:
> The continuation characters (/) should prevent this (or at least do for me under sh, bash, zsh and fsh...). I did spot a missing / in the elastic  instructions.

It seems git hub places an extra end of line character at the end.  I still have to add an extra (\\) at the end to prevent automatic execution.

Reply:
Are you looking at the rendered one here https://github.com/Materials-Consortia/optimade-python-tools/blob/78685f5d6d859e6ed635174689b03bfd7a5807bd/INSTALL.md? I did add an extra `/` for elastic in my last commits, as I said above

Reply:
Yes, I looked at the rendered version.

Reply:
> 3 It seems to hang in models/test_structures.py::test_bad_structures it also hangs when I run only test_structures.py (There are two files named test_structures hence the confusion.) I was still running it using my code branch. So perhaps my changes in mongo.py broke something. I'll look into this tomorrow.

It turned out to be a problem with the conda environment. Removing the old environment and creating a new one solved the problem. I still have no idea why the original environment stopped functioning.